File: src/main/java/com/api/parkingcontrol/repository/ParkingSpotModelRepository.java
package com.api.parkingcontrol.repository;

import java.util.UUID;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import com.api.parkingcontrol.model.ParkingSpotModel;

@Repository
public interface ParkingSpotModelRepository extends JpaRepository<ParkingSpotModel, UUID> {
    boolean existsByLicensePlateCar(String licensePlateCar);

    boolean existsByParkingSpotNumber(String parkingSpotNumber);

    boolean existsByApartmentAndBlock(String apartment, String block);
}

File: src/main/kotlin/com/api/parkingcontrol/repository/ParkingSpotModelRepository.kt
package com.api.parkingcontrol.repository

import com.api.parkingcontrol.model.ParkingSpotModel
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface ParkingSpotModelRepository : JpaRepository<ParkingSpotModel, UUID> {
    fun existsByLicensePlateCar(licensePlateCar: String?): Boolean

    fun existsByParkingSpotNumber(parkingSpotNumber: String?): Boolean

    fun existsByApartmentAndBlock(apartment: String?, block: String?): Boolean
}